src/v/iceberg/rest_client/types.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "base/seastarx.h"
+
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/sstring.hh>
+
+#include <ada/expected.h>
+
+namespace iceberg::rest_client {
+
+struct json_parse_error {
+    ss::sstring context;
+    ss::sstring error;
+};
+
+// The domain error represents the sum of all error types which can be
+// encountered during rest-client operations.
+using domain_error = std::variant<ss::sstring, json_parse_error>;
```

Comment:
I added it early in the PR to represent errors which did not fit into other categories, now we have types for most errors that we can encounter, I think it can be removed.

Reply:
removed